src/coreclr/jit/gentree.cpp

```diff
@@ -5175,11 +5175,13 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
 
             case GT_CNS_LNG:
             case GT_CNS_INT:
-                // TODO-WASM: needs tuning based on the [S]LEB128 encoding size.
-                NYI_WASM("GT_CNS_LNG/GT_CNS_INT costing");
-                costEx = 0;
-                costSz = 0;
+            {
+                GenTreeIntConCommon* con = tree->AsIntConCommon();
+                ssize_t              imm = con->IconValue();
+                costEx                   = 1;
+                costSz                   = 1 + (int)(tree->IsUnsigned() ? SizeOfULEB128(imm) : SizeOfSLEB128(imm));
```

Comment:
```suggestion
                costSz                   = 1 + static_cast<int>(SizeOfSLEB128(imm));
```
Constants are never "unsigned".